apps/ios/SimpleX SE/ShareModel.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  ShareModel.swift
+//  SimpleX SE
+//
+//  Created by Levitating Pineapple on 09/07/2024.
+//  Copyright © 2024 SimpleX Chat. All rights reserved.
+//
+
+import UniformTypeIdentifiers
+import SwiftUI
+import Combine
+import SimpleXChat
+
+class ShareModel: ObservableObject {
+    @Published var item: NSExtensionItem?
+    @Published var chats = Array<ChatData>()
+    @Published var decodedImages = Dictionary<Int, UIImage>()
+    @Published var search = String()
+    @Published var comment = String()
+    @Published var selected: ChatData?
+    @Published var isLoaded = false
+    @Published var progress: Double?
+    @Published var error: ShareError?
+
+    var completion: (() -> Void)?
+    private let chatsSubject = PassthroughSubject<Array<ChatData>, Never>()
+    private var bag = Set<AnyCancellable>()
+
+    init() {
+        // Attempt loading chats
+        Task {
+            switch fetchChats() {
+            case let .success(chats):
+                await MainActor.run {
+                    self.chatsSubject.send(chats)
+                    withAnimation { self.isLoaded = true }
+                }
+            case let .failure(error):
+                await MainActor.run { self.error = error }
+            }
+        }
+
+        // Binding: Filter chats based on debounced search query
+        $search
+            .removeDuplicates()
```

Comment:
why do we remove duplicates here?

Reply:
To avoid doing unnecessary computations.
But yes - this probably could be done after throttling.